src/services/FirebaseService.ts

```diff
@@ -0,0 +1,100 @@
+import 'firebase/auth';
+
+import firebase from 'firebase/app';
+
+const config = {
+  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
+  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.REACT_APP_FIREBASE_APP_ID,
+};
+
+class Firebase {
+  private auth: firebase.auth.Auth;
+
+  constructor() {
+    if (!firebase.apps.length) {
+      firebase.initializeApp(config);
+    } else {
+      firebase.app(); // if already initialized, use that one
+    }
+    this.auth = firebase.auth();
+  }
+
+  getAuth(): firebase.auth.Auth {
+    return this.auth;
+  }
+
+  // *** Auth API ***
+  // Creates a user and signs the user in
+  doCreateUserWithEmailAndPassword(
+    email: string,
+    password: string
+  ): Promise<firebase.auth.UserCredential> {
+    return this.auth.createUserWithEmailAndPassword(email, password);
+  }
+
+  doSignInWithEmailAndPassword(
+    email: string,
+    password: string,
+    staySignedIn: boolean
+  ): Promise<firebase.auth.UserCredential> {
+    if (staySignedIn) {
+      this.auth.setPersistence(firebase.auth.Auth.Persistence.LOCAL);
+    } else {
+      this.auth.setPersistence(firebase.auth.Auth.Persistence.SESSION);
+    }
```

Comment:
Do you mind explaining how this will work?

Reply:
So session means the user will be logged out after the window is closed, and local means the user will stay signed in when the window is closed and re-opened.